### PR TITLE
feat(pagination): sort by weight

### PR DESCRIPTION
**Pull request by Geobert**
added `weight` field in frontmatter in order to tackle https://github.com/cobalt-org/cobalt.rs/issues/671

### CODE REVIEW

**Comment by epage** [comment]
Should the default sort_by be updated to be weight, then published_date?

**Comment by Geobert** [comment]
> 
> 
> Should the default sort_by be updated to be weight, then published_date?

This is a great idea, I'll add this to this PR right now :)

File: src/cobalt_model/pagination_config.rs [hunk 1]
```diff
@@ -4,8 +4,11 @@ use std::vec::Vec;
 use super::SortOrder;
 
 pub const DEFAULT_PERMALINK_SUFFIX: &str = "{{num}}/";
-pub const DEFAULT_SORT: &str = "published_date";
+
 pub const DEFAULT_PER_PAGE: i32 = 10;
+lazy_static! {
```
Comment (epage) [inline src/cobalt_model/pagination_config.rs]: Reminds me that I need to go back sometime and move from lazy_static to once_cell

File: src/cobalt_model/pagination_config.rs [hunk 2]
```diff
@@ -150,7 +153,7 @@ impl PaginationConfigBuilder {
         let permalink_suffix =
             permalink_suffix.unwrap_or_else(|| DEFAULT_PERMALINK_SUFFIX.to_owned());
         let order = order.unwrap_or(SortOrder::Desc);
-        let sort_by = sort_by.unwrap_or_else(|| vec![DEFAULT_SORT.to_owned()]);
+        let sort_by = sort_by.unwrap_or_else(|| DEFAULT_SORT.to_vec());
```
Comment (epage) [inline src/cobalt_model/pagination_config.rs]: I think `clone` would better convey intent but oh well

Comment (Geobert) [reply src/cobalt_model/pagination_config.rs]: I can change it, I love having your opinion on this kind of stuff, it teach me to better express the intent of my code :)

File: src/cobalt_model/pagination_config.rs (hunk 1 above)
Comment (Geobert) [reply src/cobalt_model/pagination_config.rs]: Just discovering `once_cell`, looks really great!

**Pull request merged**